app/components/event-card.js

```diff
@@ -22,16 +22,23 @@ export default Component.extend({
     this._super(...arguments);
     const $innerSpan = this.$('.header > span');
     const $header = this.$('.header');
-    while ($innerSpan.outerHeight() > $header.height()) {
-      $header.attr('data-content', $innerSpan.text());
-      $header.attr('data-variation', 'tiny');
-      $innerSpan.text((index, text) => {
+    this.ellipsisedPopupElement($header, $innerSpan);
+    const $descriptionSpan = this.$('.description > span');
+    const $description = this.$('.description');
+    this.ellipsisedPopupElement($description, $descriptionSpan);
+  },
+
+  ellipsisedPopupElement(description, descriptionSpan) {
+    description.attr('data-content', descriptionSpan.text());
+    description.attr('data-variation', 'tiny');
+    while (descriptionSpan.outerHeight() > description.height()) {
+      descriptionSpan.text((index, text) => {
         return text.replace(/\W*\s(\S)*$/, '...');
       });
-      $header.popup({
+      description.popup({
         position: 'top center'
       });
-      this.set('$header', $header);
+      this.set('description', description);
```

Comment:
@harshitagupta30 This seems incorrect as this would simply set `description` to the passed description element for header also, please make it generic.